### DISCUSSION TITLE
Accessing o1 models

### DISCUSSION THREAD

**Discussion by chris-sanders** [open-webui/open-webui#7243] [Q&A] [answered]
I'm using pipelines and the openai_manifold_pipeline because that's what the docs seem to imply is the expected way to setup providers. In general all of my providers are working fine. However, the o1 models aren't listed. I'm accessing those via litellm which works fine, but I run litellm _just_ to access the o1 models.

1. Is the manifold pipelines the recommend configuration, I'm starting to question that since the o1 models never showed up.
2. What is the preferred method to get direct access to the o1 models.

I'm running in Kubernetes with the helm chart, and I'm auto-updating to the latest releases. I don't think any of that should matter much for this question but I'm anticipate the question anyway.

**Comment by mowtschan** [comment]
I guess OpenAI released 1o models for  APIs, so I don't use any pipeline:
<img width="371" alt="image" src="https://github.com/user-attachments/assets/1a181c3d-d5fb-469a-9dd6-980b32433fd5">

**Reply by chris-sanders** [reply]
Can you point me in the direction of how you're configured then?
Do i just drop the openai pipeline and configure it somewhere else instead? I just need pipelines for the other providers like Anthropic?

Are you saying you just have OpenAI configured directly, but then you can't use pipelines or can I somehow configure both? I don't want to loose access to Anthropic / Gemini which is why I have pipelines configured right now.

**Comment by mowtschan** [comment, marked as answer]
1. OpenAI:
just set you API key in Admin Panel -> Settings -> Connection
![openai](https://github.com/user-attachments/assets/15ce5735-7088-4688-a24a-6f55172999b0)

2. Anthropic:
just use Function, don't need to use pipeline
  2.1) get it here: https://openwebui.com/f/justinrahb/anthropic/
  2.2) add your API key (Valves)
![anthropic](https://github.com/user-attachments/assets/fcfddc6e-63e4-425d-bb58-bb5ae60ecb41)

3. Gemini:
similar to Anthropic:
  3.1) get it here: https://openwebui.com/f/justinrahb/google_genai/
  3.2) add your API key

**Reply by chris-sanders** [reply]
Great thanks! I'll try setting up functions instead.
I'll mark this is as answered.
